CMakeLists.txt

```diff
@@ -257,6 +282,20 @@ target_include_directories(fmt-header-only INTERFACE
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:${FMT_INC_DIR}>)
 
+if (FMT_CAN_MODULE AND FMT_TEST)
+  set (tstmod "modfmt")
+  add_library(${tstmod} STATIC ${FMT_INTERFACE_UNIT} ${FMT_HEADERS})
```

Comment:
The additional module besides the traditional library.
I want to move this target into the other `CMakeLists.txt` in the `test` subdirectory. Hence the idea of generating the required compiler-specific compile options by CMake functions rather than hardcoding them. If at all possible, I want to turn the test-module target into a `OBJECT` library if there is a way to introduce a dependency from the test-module-related object artifact to the invocation of the **compile** step of the module tests where the BMI must be available for consumption. So far I haven't found a way to express this. Either the syntax doesn't provide a way to get at the path to the object artifact, or CMake complains about a dependency on the BMI artifact. In the latter case it doesn't accept it because it either has no clue about how to create this artifact or it simply refuses to accept the file extension. In the former case there doesn't seem to exist a reliable and portable way to get at the full path, CMake generator expressions aren't supported in the required syntactical constructs. I need a "proxy" for the actual BMI and this is either a library name at a known path that I can then reliably resynthesize, or the object artifact with the unknown path. So far I go with this test-only library.

Reply:
I don't think we should be building another module from the same source. Not doing this also eliminates the need in the `bmi_vars` function.

Reply:
This is not about compiling "another module". It is about compiling a module at all to enable testing the module-related stuff. The test must be conducted in "traditional lib" mode to make all of the existing tests work. {fmt} is compiled as it ever was. Because the module-based tests cannot excercise those parts of the current functional tests that need access to internal, non-exported parts of the library, {fmt} better be compiled so for testing. Therefore the need for a test-only additional module binary hidden away in the test subdirectory. And because it can no longer clash with the  Real Stuff® in the test's root directory the name of the binary is fixed now.

Reply:
Makes sense now, thanks for clarification. I would recommend adding a comment explaining this.